salt/states/dockerng.py

```diff
@@ -2145,7 +2145,7 @@ def volume_present(name, driver=None, driver_opts=None):
             return ret
     # volume exits, check if driver is the same.
     volume = volumes[0]
-    if volume['Driver'] != driver:
+    if driver is not None and volume['Driver'] != driver:
```

Comment:
I think it is better to initiate `driver` parameter with the default docker's volume driver name.

```python
if driver is None:
  driver = get_default_docker_driver()
```
Otherwise when one removes `- driver: my driver` from the sls file, the change to default driver won't be detected.